app/battle/fight.py

```diff
@@ -0,0 +1,12 @@
+from app.knights.knights import Knight
+
+
+def fight(knight1: Knight, knight2: Knight) -> None:
+    knight1.hp -= knight2.power - knight1.protection
+    knight2.hp -= knight1.power - knight2.protection
```

Comment:
Similarly, ensure that `knight1.power - knight2.protection` does not result in negative damage. This could inadvertently heal `knight2` if not handled properly.